pkg/synopsysctl/cmd_serve_UI.go

```diff
@@ -0,0 +1,652 @@
+	}
+	polarisObj.Licenses.Coverity = data
+
+	return &polarisObj, nil
+}
+
+// convertPolarisObjToUIResponse converts the Polaris Object Specification into the Response format
+// that can be sent to the User Interface
+func convertPolarisObjToUIResponse(polarisObj polaris.Polaris) (*PolarisUIRequestResponse, error) {
+	// Populate Polaris Config Fields
+	polarisUIRequestConfig := &PolarisUIRequestResponse{}
+	polarisUIRequestConfig.Namespace = polarisObj.Namespace
+
+	polarisUIRequestConfig.Version = polarisObj.Version
+	polarisUIRequestConfig.EnvironmentDNS = polarisObj.EnvironmentDNS
+	polarisUIRequestConfig.ImagePullSecrets = polarisObj.ImagePullSecrets
+	// TODO - Postgres host and port are not supported
+	// polarisUIRequestConfig.PostgresHost = polarisObj.PolarisDBSpec.PostgresDetails.Host
+	// polarisUIRequestConfig.PostgresPort  = strconv.FormatInt(polarisObj.PolarisDBSpec.PostgresDetails.Port)
+	polarisUIRequestConfig.PostgresUsername = polarisObj.PolarisDBSpec.PostgresDetails.Username
+	polarisUIRequestConfig.PostgresPassword = polarisObj.PolarisDBSpec.PostgresDetails.Password
+	polarisUIRequestConfig.PostgresSize = polarisObj.PolarisDBSpec.PostgresDetails.Storage.StorageSize
+
+	polarisUIRequestConfig.SMTPHost = polarisObj.PolarisDBSpec.SMTPDetails.Host
+	polarisUIRequestConfig.SMTPPort = strconv.FormatInt(int64(polarisObj.PolarisDBSpec.SMTPDetails.Port), 10)
+	polarisUIRequestConfig.SMTPUsername = polarisObj.PolarisDBSpec.SMTPDetails.Username
+	polarisUIRequestConfig.SMTPPassword = polarisObj.PolarisDBSpec.SMTPDetails.Password
+	polarisUIRequestConfig.SMTPSenderEmail = polarisObj.PolarisDBSpec.SMTPDetails.SenderEmail
+
+	polarisUIRequestConfig.UploadServerSize = polarisObj.PolarisDBSpec.UploadServerDetails.Storage.StorageSize
+	polarisUIRequestConfig.EventstoreSize = polarisObj.PolarisDBSpec.EventstoreDetails.Storage.StorageSize
+	polarisUIRequestConfig.MongoDBSize = polarisObj.PolarisDBSpec.MongoDBDetails.Storage.StorageSize
+	polarisUIRequestConfig.DownloadServerSize = polarisObj.PolarisSpec.DownloadServerDetails.Storage.StorageSize
+
+	polarisUIRequestConfig.EnableReporting = polarisObj.EnableReporting
+	polarisUIRequestConfig.ReportStorageSize = polarisObj.ReportingSpec.ReportStorageDetails.Storage.StorageSize
+
+	polarisUIRequestConfig.OrganizationDescription = polarisObj.OrganizationDetails.OrganizationProvisionOrganizationDescription
+	polarisUIRequestConfig.OrganizationName = polarisObj.OrganizationDetails.OrganizationProvisionOrganizationName
+	polarisUIRequestConfig.OrganizationAdminName = polarisObj.OrganizationDetails.OrganizationProvisionAdminName
+	polarisUIRequestConfig.OrganizationAdminUsername = polarisObj.OrganizationDetails.OrganizationProvisionAdminUsername
+	polarisUIRequestConfig.OrganizationAdminEmail = polarisObj.OrganizationDetails.OrganizationProvisionAdminEmail
+	polarisUIRequestConfig.CoverityLicensePath = "" // User needs to provide License Path every time
```

Comment:
fine for now, probably can make it so if valid license is already in the secret, maybe don't require to put the path in again.